validate_sample.py

```diff
@@ -0,0 +1,339 @@
+    global error_count
+    column = 'ARID'
+    maximum_length = 21
+    if _check_mandatory_value_exists(line_number, column, mandatory=True, sample_row=sample_row):
+        _check_length(column, sample_row[column], line_number, maximum_length)
+    if sample_row[column] in ARID:
+        print(f'Line {line_number}: {column}: {sample_row[column]} is duplicated in sample file.')
+        error_count += 1
+    else:
+        ARID.add(sample_row[column])
+
+
+def validate_estab_arid(line_number, sample_row):
+    column = 'ESTAB_ARID'
+    maximum_length = 21
+    if _check_mandatory_value_exists(line_number, column, mandatory=True, sample_row=sample_row):
+        _check_length(column, sample_row[column], line_number, maximum_length)
+
+
+def validate_uprn(line_number, sample_row):
+    global error_count
+    column = 'UPRN'
+    maximum_length = 12
+    if _check_mandatory_value_exists(line_number, column, mandatory=False, sample_row=sample_row):
+        _check_length(column, sample_row[column], line_number, maximum_length)
+        uprn = sample_row[column]
+        if not uprn.isnumeric() and uprn != '':
+            print(f'Line {line_number}: {column}: {sample_row[column]} is not a valid integer.')
+            error_count += 1
+
+
+def validate_address_type(line_number, sample_row):
+    global error_count
+    column = 'ADDRESS_TYPE'
+    if _check_mandatory_value_exists(line_number, column, mandatory=True, sample_row=sample_row):
+        address_type = sample_row[column]
+        if address_type not in {"HH", "CE", "SPG"}:
+            print(f'Line {line_number}: {column}: {sample_row[column]} is not valid.')
+            error_count += 1
+
+
+def validate_estab_type(line_number, sample_row):
+    column = 'ESTAB_TYPE'
+    if _check_mandatory_value_exists(line_number, column, mandatory=True, sample_row=sample_row):
+        _is_valid_estab_type(line_number, sample_row[column])
+
+
+def validate_address_level(line_number, sample_row):
+    global error_count
+    column = 'ADDRESS_LEVEL'
+    if _check_mandatory_value_exists(line_number, column, mandatory=True, sample_row=sample_row):
+        if sample_row[column] not in {"E", "U"}:
+            print(f'Line {line_number}: {column}: {sample_row[column]} is not valid.')
+            error_count += 1
+
+
+def validate_abp_code(line_number, sample_row):
+    column = 'ABP_CODE'
+    maximum_length = 6
+    if _check_mandatory_value_exists(line_number, column, mandatory=True, sample_row=sample_row):
+        _check_length(column, sample_row[column], line_number, maximum_length)
+
+
+def validate_org_name(line_number, sample_row):
+    column = 'ORGANISATION_NAME'
+    maximum_length = 60
+    if _check_mandatory_value_exists(line_number, column, mandatory=False, sample_row=sample_row):
+        _check_length(column, sample_row[column], line_number, maximum_length)
+
+
+def validate_address_line(line_number, sample_row):
+    maximum_length = 60
+    # only address line 1 is mandatory
```

Comment:
Other address lines should still be validated for max length?

Reply:
The other address lines are already validated for line length (they're all max 60 in the spec).